### ISSUE TITLE
Xml Read erro

### ISSUE THREAD

**Issue by dragoso** [FasterXML/jackson-dataformat-xml#421]
Some unread xmls correctly, for example:

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<imgdir name="2112007.img">
	<imgdir name="info">
		<string name="info" value="&amp;#54028;&amp;#50892;&amp;#50648;&amp;#47533;&amp;#49436;"/>
	</imgdir>
	<imgdir name="0">
		<canvas name="0" width="86" height="63">
			<vector name="origin" x="43" y="63"/>
			<int name="z" value="0"/>
		</canvas>
		<imgdir name="event">
			<imgdir name="0">
				<int name="type" value="0"/>
				<int name="state" value="1"/>
			</imgdir>
		</imgdir>
		<imgdir name="hit">
			<canvas name="0" width="124" height="73">
				<vector name="origin" x="61" y="73"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
			<canvas name="1" width="162" height="89">
				<vector name="origin" x="81" y="89"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
			<canvas name="2" width="182" height="99">
				<vector name="origin" x="95" y="99"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
		</imgdir>
	</imgdir>
	<imgdir name="1">
		<canvas name="0" width="86" height="61">
			<vector name="origin" x="43" y="61"/>
			<int name="z" value="0"/>
		</canvas>
		<imgdir name="event">
			<imgdir name="0">
				<int name="type" value="0"/>
				<int name="state" value="2"/>
			</imgdir>
		</imgdir>
		<imgdir name="hit">
			<canvas name="0" width="134" height="61">
				<vector name="origin" x="56" y="60"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
			<canvas name="1" width="172" height="80">
				<vector name="origin" x="77" y="79"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
			<canvas name="2" width="192" height="88">
				<vector name="origin" x="93" y="87"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
		</imgdir>
	</imgdir>
	<imgdir name="2">
		<canvas name="0" width="97" height="47">
			<vector name="origin" x="43" y="46"/>
			<int name="z" value="0"/>
		</canvas>
		<imgdir name="event">
			<imgdir name="0">
				<int name="type" value="0"/>
				<int name="state" value="3"/>
			</imgdir>
		</imgdir>
		<imgdir name="hit">
			<canvas name="0" width="111" height="56">
				<vector name="origin" x="52" y="55"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
			<canvas name="1" width="135" height="62">
				<vector name="origin" x="72" y="61"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
			<canvas name="2" width="144" height="70">
				<vector name="origin" x="75" y="69"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
		</imgdir>
	</imgdir>
	<imgdir name="3">
		<canvas name="0" width="83" height="33">
			<vector name="origin" x="36" y="32"/>
			<int name="z" value="0"/>
		</canvas>
		<imgdir name="event">
			<imgdir name="0">
				<int name="type" value="0"/>
				<int name="state" value="4"/>
			</imgdir>
		</imgdir>
		<imgdir name="hit">
			<canvas name="0" width="108" height="52">
				<vector name="origin" x="58" y="52"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
			<canvas name="1" width="141" height="77">
				<vector name="origin" x="72" y="77"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
			<canvas name="2" width="97" height="58">
				<vector name="origin" x="42" y="58"/>
				<int name="z" value="0"/>
				<int name="delay" value="120"/>
			</canvas>
		</imgdir>
	</imgdir>
	<imgdir name="4">
		<canvas name="0" width="1" height="1">
			<vector name="origin" x="0" y="0"/>
		</canvas>
	</imgdir>
	<string name="action" value="boxItem0"/>
</imgdir>
```

This xml has one "imgdir" and into have 6 "imgdir" s ... but if i try, this return 2 values

**Comment by cowtowncoder** [comment]
Unfortunately this is missing almost everything needed to reproduce the issue; specifically:

1. POJOs content would be read as.
2. Jackson version being used.

I suspect this might be one of over 20 fixes that will go in 2.12 (see https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.12). But at very least make sure to try it with already released 2.11.2 if using an older version; it contains some fixes over previous versions, and especially some for handling nested `List` structures.

**Comment by cowtowncoder** [comment]
No reproduction, can not reproduce. Presumed to be fixed as per earlier notes; may be reopened with reproduction.
